lib/internal/util/inspect.js

```diff
@@ -176,10 +176,10 @@ const kArrayType = 1;
 const kArrayExtrasType = 2;
 
 /* eslint-disable no-control-regex */
-const strEscapeSequencesRegExp = /[\x00-\x1f\x27\x5c\x7f-\x9f]/;
-const strEscapeSequencesReplacer = /[\x00-\x1f\x27\x5c\x7f-\x9f]/g;
-const strEscapeSequencesRegExpSingle = /[\x00-\x1f\x5c\x7f-\x9f]/;
-const strEscapeSequencesReplacerSingle = /[\x00-\x1f\x5c\x7f-\x9f]/g;
+const strEscapeSequencesRegExp = /[\x00-\x1f\x27\x5c\x7f-\x9f]|[\ud800-\udbff](?![\udc00-\udfff])|(?<![\ud800-\udbff])[\udc00-\udfff]/;
```

Comment:
I wonder if this can be reused

https://github.com/nodejs/node/blob/265a47d680077945e49578f4cb0decc772ff8058/lib/internal/util.js#L58

Reply:
In theory, yes. I could write it as string and create a new regular expression. That way they could share the same source. Would you favor that?

Reply:
No strong opinion, I'm fine either way.

Reply:
Let's keep it for now.